PyInstaller/building/api.py

```diff
@@ -720,17 +721,20 @@ def assemble(self):
 
         if is_linux:
             # Linux: append data into custom ELF section using objcopy.
+            # For huge executables, this can take so long that users think it's died. Include a crude progress indicator
+            # so that they know otherwise.
             logger.info("Appending %s to custom ELF section in EXE", append_type)
-            retcode, stdout, stderr = exec_command_all(
-                'objcopy', '--add-section', 'pydata=%s' % append_file, build_name
-            )
-            logger.debug("objcopy returned %i", retcode)
-            if stdout:
-                logger.debug(stdout)
-            if stderr:
-                logger.debug(stderr)
-            if retcode != 0:
-                raise SystemError("objcopy Failure: %s" % stderr)
+            cmd = ['objcopy', '--add-section', f'pydata={append_file}', build_name]
+            p = subprocess.Popen(cmd, stderr=subprocess.STDOUT, stdout=subprocess.PIPE, universal_newlines=True)
+
+            target_size = os.stat(append_file).st_size + os.stat(build_name).st_size
+            while p.returncode != 0:
+                try:
+                    if p.wait(5):
+                        raise SystemError(f"objcopy Failure: {p.returncode} {p.stdout.read()}")
+                except subprocess.TimeoutExpired:
+                    logger.info(f"objdump-ed {os.stat(build_name).st_size // 1_000} out of {target_size // 1_000} kB")
```

Comment:
It's `objcopy`, not `objdump`. But perhaps "Copied X out of Y kB" or "Appended X out of Y kB" might be better?